src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +122,46 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer();
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string) {
+        if (gatewayId === PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS) {
+            this._pollingInterval = window.setTimeout(async () => {
+                try {
+                    this._pollingTimer += POLLING_INTERVAL;
+                    const res = await this._paypalCommerceRequestSender.getOrderStatus();
+                    if (res.status !== ORDER_STATUS_CREATED && res.status !== ORDER_STATUS_APPROVED) {
+                        this._deinitializePollingTimer();
+                    }
+                    if (res.status === ORDER_STATUS_APPROVED) {
+                        this._deinitializePollingTimer();
+                        this._tokenizePayment({ orderID: this._paypalCommercePaymentProcessor.getOrderId() }, submitForm);
+                    } else {
+                        if (this._pollingTimer < POLLING_MAX_TIME) {
+                            this._initializePollingMechanism(submitForm, gatewayId);
+                        } else {
+                            this._deinitializePollingTimer();
+
+                            return;
```

Comment:
You don't need a `return` as there's no code after this line that can be executed in this code block.

Reply:
removed